packages/lodestar-validator/src/api/impl/rest/beacon/beacon.ts

```diff
@@ -1,14 +1,36 @@
-import {Bytes32, Fork, SyncingStatus, BeaconBlock, BeaconState, Number64, Uint64} from "@chainsafe/lodestar-types";
+import {
+  BeaconBlock,
+  BeaconState,
+  BLSPubkey,
+  Bytes32,
+  Fork,
+  Number64,
+  SyncingStatus,
+  Uint64,
+  ValidatorResponse
+} from "@chainsafe/lodestar-types";
 import {IBeaconApi} from "../../../interface/beacon";
 import {HttpClient} from "../../../../util";
 import {ILogger} from "@chainsafe/lodestar-utils/lib/logger";
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {toHexString} from "@chainsafe/ssz";
 
 export class RestBeaconApi implements IBeaconApi {
 
-  private client: HttpClient;
+  private readonly client: HttpClient;
 
-  public constructor(restUrl: string, logger: ILogger) {
-    this.client = new HttpClient({urlPrefix: `${restUrl}/node`}, {logger});
+  private readonly config: IBeaconConfig;
+
+  public constructor(config: IBeaconConfig, restUrl: string, logger: ILogger) {
+    this.client = new HttpClient({urlPrefix: `${restUrl}/validator`}, {logger});
+    this.config = config;
+  }
+
+
+  public async getValidator(pubkey: BLSPubkey): Promise<ValidatorResponse|null> {
```

Comment:
Why do we actually need the `ValidatorResponse` type?
The api seems to be designed to NOT involve `ValidatorIndex` (anywhere other than this addition), but instead rely on passing `BLSPubkey` to identify the validator.

Reply:
I see, this is a general beacon API. Can we use the same shape lighthouse is using? https://lighthouse-book.sigmaprime.io/http_beacon.html#beaconvalidators

Reply:
> Why do we actually need the `ValidatorResponse` type?
> The api seems to be designed to NOT involve `ValidatorIndex` (anywhere other than this addition), but instead rely on passing `BLSPubkey` to identify the validator.

I was hoping we will stick with pubkey as identifier. But to construct AggregateAndProof, we need index which is a bit of a pickle.

> I see, this is a general beacon API. Can we use the same shape lighthouse is using? https://lighthouse-book.sigmaprime.io/http_beacon.html#beaconvalidators

I like that response format but I api is not really rest like:
- GET /validators?pubkeys=[]&stateRoot=
- GET /validators/{pubkey}
would be more aligned with rest conventions

Reply:
:+1:  Yeah I was more just talking about the response format